datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -622,8 +714,8 @@ private boolean getCustomCheckpointing(DatastreamTask task) {
     boolean customCheckpointing = _connectors.get(task.getConnectorType()).isCustomCheckpointing();
 
     Datastream datastream = task.getDatastreams().get(0);
-    if (datastream.hasMetadata()
-        && datastream.getMetadata().containsKey(DatastreamMetadataConstants.CUSTOM_CHECKPOINT)) {
+    if (datastream.hasMetadata() && datastream.getMetadata() != null &&
```

Comment:
Can we use Objects.requireNonNull() in place of `datastream.getMetadata() != null`?